rust/lance-table/src/format/manifest.rs

```diff
@@ -662,9 +662,27 @@ pub enum VersionPart {
     Patch,
 }
 
+fn bump_version(version: &mut semver::Version, part: VersionPart) {
+    match part {
+        VersionPart::Major => {
+            version.major += 1;
+            version.minor = 0;
+            version.patch = 0;
+        }
+        VersionPart::Minor => {
+            version.minor += 1;
+            version.patch = 0;
+        }
+        VersionPart::Patch => {
+            version.patch += 1;
+        }
+    }
+}
+
 impl WriterVersion {
     /// Try to parse the version string as a semver string. Returns None if
     /// not successful.
+    #[deprecated(note = "Use `lance_lib_version()` instead")]
```

Comment:
I was thinking this could still be useful for non Lance library version as long as they are using semver. But this should be fine to deprecate for now.

Reply:
Yeah if we start doing carveouts for other libraries, we can add that back.